design-proposals/golden-image-delivery-and-update-pipeline.md

```diff
@@ -0,0 +1,275 @@
+kind: DataVolume
+metadata:
+   name: my-fedora-vm-image
+spec:
+  pvc:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 2Gi
+  dataSource:
+    name: fedora
+    namespace: golden-images 
+```
+
+## DataImportCron Example
+
+This example polls for image tags for the quay.io/kubevirt/fedora image and imports the latest image sorting tags based on semver. As new tags are introduced, the DataImportCron will import them as new PVCs and automatically manage updating the corresponding DataSource.
+
+* **source** specifies where to poll from
+* **managedDataSource** specifies the name of the corresponding DataSource this cron will manage. The controller should create this DataSource if it doesn't exist, and adopt ownership of the DataSource if it does already exist.
+* **schedule** specifies in cron format when and how often to look for new imports 
+* **garbageCollectOutdated** specifies whether old PVCs should be cleaned up after a new PVC is imported.
+
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataImportCron
+metadata:
+  name: fedora-image-import-cron
+  namespace: golden-images
+spec:
+  source:
+    registry:
+      url: docker://quay.io/kubevirt/fedora
+  schedule: "0 */12 * * *"
+  garbageCollectOutdated: true
+  managedDataSource: fedora
+status:
+  lastImportedPVC: quay.io/kubevirt/fedora:33
+```
+
+## DataSource Example
+
+This example shows how a DataSource uses a ready condition to indicate that the corresponding pvc exists and is populated. 
+
+The UI can use the ready condition to determine if a boot source exists for a DataSource or not. 
+
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataSource
+metadata:
+  name: fedora
+spec:
+  source:
+    pvc:
+      namespace: golden-images 
+      name: fedora-32-001
+status:
+  conditions:
+  - status: "True"
+    type: Ready
+
+```
+
+## SSP Operator CR Example
+
+This example shows how the SSP operator could expose opting into automatic image delivery and updates for images associated with their common templates. 
+
+By default, the ssp operator would simply create dangling DataSources to match the PVC sources used in the DataVolumeTemplates in the common templates. Users would import their own disks which would automatically bind to the dangling DataSources and make them linked (reflected by ready condition set to true) to a PVC. This matches the behavior that exists today.
+
+By specifying a DataImportCron on the SSP CR, one can opt-in to having DataSources automatically linked to imported PVCs and have that linkage update as new versions of the corresponding image are imported.
+
+```yaml
+apiVersion: ssp.kubevirt.io/v1beta1
+kind: SSP
+metadata:
+  name: ssp-cr
+spec: 
+  DataImportCronTemplates:
+  - metadata:
+      name: fedora-image-cron
+      namespace: golden-image-namespace
+    spec:
+      source:
+        registry:
+          url: docker://quay.io/kubevirt/fedora
+      managedDataSource: fedora
```

Comment:
I wonder if that should also be a proper reference (namespace, name).
And having the word DataSource in the field explicitly locks us down on the relationship. So it could just be "manages" or "updates" but on the other hand it's less speaking :-/

Reply:
For now, i think the `managedDataSource` should be assumed to exist in the same namespace as the cron